Use arbitrary length integers to timestamp snapshots

pbartholomew08 opened the pull request:
This will give snapshot files of the form

ux1
ux2
ux3
...
uxN

without zero padding, currently checkpoints/statistics have not been considered.

pbartholomew08 committed:
Use arbitrary length integers to timestamp snapshots

mathrack commented on src/visu.f90:
```diff
@@ -320,7 +322,8 @@ subroutine end_snapshot(itime, num)
         write(fmt4,'("(A,F16.12)")')
 
-        open(newunit=is,file="./data/snap"//trim(num)//".ini",action='write',status='replace')
+        write(filename, '(A, I0, A)') "./data/snap", num, ".ini"
```
This statement allocates the character array "filename" ?

pbartholomew08 replied:
I need to double check this one - it didn't cause any issues though

mathrack commented on src/visu.f90:
```diff
@@ -576,22 +581,23 @@ subroutine write_field(f1, pathname, filename, num, skip_ibm, flush)
-    character(len=(len(pathname) + 1 + len(varname) + 1 + len(num) + 1 + len(ext))) :: gen_filename
-    write(gen_filename, "(A)") pathname//'/'//varname//'-'//num//'.'//ext
+    character(len=:), allocatable :: gen_filename
+    gen_filename = pathname//'/'//varname//'-'//int_to_str(num)//'.'//ext
```
This statement allocates the character array "gen_filename" ?

pbartholomew08 replied:
Yes - allocatable intrinsic types are (re)allocated on assignment (F2003).

mathrack replied:
Nice

mathrack commented on src/visu.f90:
```diff
@@ -601,17 +607,25 @@ end function gen_filename
-    character(len=(len(path_to_h5file) + len(num) + 1+ len(filename))) :: gen_h5path
-    write(gen_h5path, "(A)") path_to_h5file//num//"/"//filename
+    character(len=:), allocatable :: gen_h5path
+    gen_h5path = path_to_h5file//int_to_str(num)//"/"//filename
```
This statement allocates the character array "gen_h5path" ?

pbartholomew08 replied:
See https://github.com/xcompact3d/Incompact3d/pull/94#discussion_r795452179

pbartholomew08 committed:
Rewrite filename string as allocation

The Fortran2003 standard (re)allocates allocatable intrinsics on
assignment - it's not clear that this would work when using in the
form

write(str, fmt) ...

it seems safer to simply

str = ...

pbartholomew08 commented:
Replaces ``write(str, fmt) ...`` with ``str = ...`` where ``str`` is an allocatable character array - it isn't clear that the former is either correct or portable. (See https://github.com/xcompact3d/Incompact3d/pull/94#discussion_r795452667)

pbartholomew08 committed:
Use arbitrary-length integers in stats file names

i.e.

umean1
umean10
umean100

etc

pbartholomew08 committed:
Remove timestamp from restart.info filename

The file contains the timestamp as part of the data, there is only
one checkpoint file, so no point in maintaining multiple restart.info
files

slaizet commented:
Paul, is this formatting compatible with Paraview?

pbartholomew08 committed:
Merge branch 'master' into dev--snapshot-I0

pbartholomew08 commented:
Some code in LES models had drifted out of consistency, merged current master into branch and tested with ParaView (20 snapshots covering single and double digits)
